springwolf-core/src/main/java/io/github/springwolf/core/asyncapi/components/ComponentsService.java

```diff
@@ -5,13 +5,15 @@
 import io.github.springwolf.asyncapi.v3.model.channel.message.MessageObject;
 import io.github.springwolf.asyncapi.v3.model.channel.message.MessageReference;
 import io.github.springwolf.asyncapi.v3.model.schema.SchemaObject;
+import jakarta.annotation.Nullable;
 
 import java.util.Map;
 
 public interface ComponentsService {
 
     Map<String, SchemaObject> getSchemas();
 
+    @Nullable
     SchemaObject resolveSchema(String schemaName);
 
     String registerSchema(SchemaObject headers);
```

Comment:
Option for the next refactoring step (so that we can get rid of resolveSchema):
```suggestion
    NamedSchemaObject registerSchema(SchemaObject headers);
```